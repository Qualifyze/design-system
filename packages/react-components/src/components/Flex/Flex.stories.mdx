import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import { Flex } from './Flex';

<Meta title="Components/Flex" component={Flex} />

# Flex

<Canvas>
  <Story name="Flex">
    <Flex sx={{ p: 4, bg: 'gray.200' }}>hello</Flex>
  </Story>
</Canvas>