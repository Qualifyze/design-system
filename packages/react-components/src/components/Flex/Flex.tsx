import { styled } from '../../stitches.config';

export const Flex = styled('div', {
  boxSizing: 'border-box',
  position: 'relative',
  display: 'flex',

  variants: {
    direction: {
      row: {
        flexDirection: 'row',
      },
      column: {
        flexDirection: 'column',
      },
      rowReverse: {
        flexDirection: 'row-reverse',
      },
      columnReverse: {
        flexDirection: 'column-reverse',
      },
    },
    align: {
      start: {
        alignItems: 'flex-start',
      },
      center: {
        alignItems: 'center',
      },
      end: {
        alignItems: 'flex-end',
      },
      stretch: {
        alignItems: 'stretch',
      },
      baseline: {
        alignItems: 'baseline',
      },
    },
    justify: {
      start: {
        justifyContent: 'flex-start',
      },
      center: {
        justifyContent: 'center',
      },
      end: {
        justifyContent: 'flex-end',
      },
      between: {
        justifyContent: 'space-between',
      },
    },
    wrap: {
      noWrap: {
        flexWrap: 'nowrap',
      },
      wrap: {
        flexWrap: 'wrap',
      },
      wrapReverse: {
        flexWrap: 'wrap-reverse',
      },
    },
    gap: {
      0: {
        gap: 0,
      },
      4: {
        gap: '$4',
      },
      8: {
        gap: '$8',
      },
      12: {
        gap: '$12',
      },
      16: {
        gap: '$16',
      },
      24: {
        gap: '$24',
      },
      32: {
        gap: '$32',
      },
      48: {
        gap: '$48',
      },
      64: {
        gap: '$64',
      },
      96: {
        gap: '$96',
      },
    },
  },
  defaultVariants: {
    direction: 'row',
    align: 'stretch',
    justify: 'start',
    wrap: 'noWrap',
  },
});
