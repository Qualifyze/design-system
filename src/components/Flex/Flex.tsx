export { Flex } from 'theme-ui';
