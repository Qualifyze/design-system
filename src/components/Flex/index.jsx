import styled from '@emotion/styled'

import Box from '../Box'

const Flex = styled(Box)({
  display: 'flex',
})

export default Flex
