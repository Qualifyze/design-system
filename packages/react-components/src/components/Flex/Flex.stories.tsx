import React from 'react';

import { Flex } from './Flex';

export default {
  title: 'Components/Flex',
  component: Flex,
};

export const basic = () => (
  <Flex css={{ padding: '$2', bc: '$indigo4' }}>Hello</Flex>
);
