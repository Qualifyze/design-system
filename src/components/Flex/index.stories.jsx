import React from 'react'
import { text } from '@storybook/addon-knobs'

import Flex from './index'

export default { title: 'Flex', component: Flex }

export const Default = () => {
  const bg = text('Background', 'primary.100')
  const m = text('Margin', '2')
  const p = text('Padding', '4')

  return <Flex sx={{ bg, m, p }}>Hola, mundo!</Flex>
}
Default.story = {
  name: 'default',
}
