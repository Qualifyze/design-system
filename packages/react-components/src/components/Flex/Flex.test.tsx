import * as React from 'react';
import { render, screen } from '../../test';

import { Flex } from './Flex';

describe('Flex', () => {
  test('renders', () => {
    render(<Flex>hola</Flex>);
    expect(screen.getByText(/hola/)).toBeInTheDocument();
  });

  test('renders with css prop', () => {
    render(<Flex css={{ padding: 1 }}>hola</Flex>);
    const F = screen.getByText(/hola/);
    expect(F).toHaveStyleRule(`padding`, `1px`);
    expect(F).toMatchSnapshot();
  });

  test('renders with sx prop', () => {
    render(<Flex sx={{ p: 1 }}>hola</Flex>);
    const F = screen.getByText(/hola/);
    expect(F).toHaveStyleRule(`padding`, `4px`);
    expect(F).toMatchSnapshot();
  });
});
