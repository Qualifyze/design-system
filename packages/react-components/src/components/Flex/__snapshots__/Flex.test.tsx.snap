// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex passes \`aria-\` attributes through to the DOM 1`] = `
<div
  aria-hidden="true"
  class="c-dwCIWy c-dwCIWy-ejCoEP-direction-row c-dwCIWy-irEjuD-align-stretch c-dwCIWy-awKDG-justify-start c-dwCIWy-kVNAnR-wrap-noWrap"
>
  hola
</div>
`;

exports[`Flex passes \`data-\` attributes through to the DOM 1`] = `
<div
  class="c-dwCIWy c-dwCIWy-ejCoEP-direction-row c-dwCIWy-irEjuD-align-stretch c-dwCIWy-awKDG-justify-start c-dwCIWy-kVNAnR-wrap-noWrap"
  data-testid="some-random-id"
>
  hola
</div>
`;

exports[`Flex renders as passed HTML element 1`] = `
<h1
  class="c-dwCIWy c-dwCIWy-ejCoEP-direction-row c-dwCIWy-irEjuD-align-stretch c-dwCIWy-awKDG-justify-start c-dwCIWy-kVNAnR-wrap-noWrap"
>
  hola
</h1>
`;

exports[`Flex renders with css prop 1`] = `
<div
  class="c-dwCIWy c-dwCIWy-ejCoEP-direction-row c-dwCIWy-irEjuD-align-stretch c-dwCIWy-awKDG-justify-start c-dwCIWy-kVNAnR-wrap-noWrap c-dwCIWy-iiMKLbU-css"
>
  hola
</div>
`;

exports[`Flex renders with theme-aware values from Stitches config 1`] = `
<div
  class="c-dwCIWy c-dwCIWy-ejCoEP-direction-row c-dwCIWy-irEjuD-align-stretch c-dwCIWy-awKDG-justify-start c-dwCIWy-kVNAnR-wrap-noWrap c-dwCIWy-ibcCYNQ-css"
>
  hola
</div>
`;
